oracle/src/main/java/org/jdbi/v3/oracle/OracleReturning.java

```diff
@@ -56,83 +57,52 @@
  * }
  * </pre>
  * Though you can bind multiple params, and whatnot
+ * <p>
+ * This class is beta, and may be changed incompatibly or removed at any time.
  */
-@Deprecated
-public class OracleReturning<ResultType> implements StatementCustomizer
-{
+public class OracleReturning<ResultType> implements StatementCustomizer {
     private final RowMapper<ResultType> mapper;
     private final List<int[]> binds = new ArrayList<>();
     private StatementContext context;
     private List<ResultType> results;
-    private Class<?> oraclePS ;
-    private Method registerReturnParameter ;
-    private Method getReturnResultSet ;
-    private Object stmt ;
+    private OraclePreparedStatement stmt;
 
     /**
      * Provide a mapper which knows how to do positional access, sadly the
      * {@link BeanMapper} uses the names in the result set
      *
      * @param mapper Must use only positional access to the result set
      */
-    public OracleReturning(RowMapper<ResultType> mapper)
-    {
+    public OracleReturning(RowMapper<ResultType> mapper) {
         this.mapper = mapper;
-        try {
-            this.oraclePS = Class.forName("oracle.jdbc.OraclePreparedStatement");
-            this.registerReturnParameter = oraclePS.getMethod("registerReturnParameter", new Class[]{int.class, int.class});
-            this.getReturnResultSet = oraclePS.getMethod("getReturnResultSet");
-        }
-        catch (Exception e) {
-            throw new RuntimeException(e);
-        }
     }
 
     /**
      * @see StatementCustomizer#beforeExecution(java.sql.PreparedStatement, StatementContext)
      */
     @Override
-    public void beforeExecution(PreparedStatement stmt, StatementContext ctx) throws SQLException
-    {
+    public void beforeExecution(PreparedStatement stmt, StatementContext ctx) throws SQLException {
         this.context = ctx;
 
-        if (!oraclePS.isAssignableFrom(stmt.getClass())) {
-            try {
-                final Method get_delegate = stmt.getClass().getMethod("getDelegate");
-                final Object candidate = get_delegate.invoke(stmt);
-                if (!oraclePS.isAssignableFrom(candidate.getClass())) {
-                    throw new Exception("Obtained delegate, but it still wasn't an OraclePreparedStatement");
-                }
-                else {
-                    this.stmt = candidate ;
-                }
-            }
-            catch (Exception e) {
-                throw new IllegalStateException("Statement is not an OraclePreparedStatement, nor" +
-                                                "one which we know how to find it from", e);
-            }
-        }
-        else {
-            this.stmt = stmt ;
+        if (!stmt.isWrapperFor(OraclePreparedStatement.class)) {
+            throw new IllegalStateException("Statement is not an instance of, nor a wrapper of, OraclePreparedStatement");
         }
+        this.stmt = stmt.unwrap(OraclePreparedStatement.class);
```

Comment:
It's worth noting the reason this class existed in the first place is because `OraclePreparedStatement` doesn't play nice with `getX(String,...)` methods. It wants the column number.

Since the reflective mappers are now all using column index, this class might not have a reason to exist any more. But we need tests to validate that assumption.